### PR TITLE
Introduce nettype passt to qemu vm

### PR DESCRIPTION
**Pull request by luckyh**
passt is designed as a replacement for slirp (--netdev user), a new
approach implemented usermode networking which improves on security
and performance.
    
This patch enables the functionality to allow qemu VMs to use passt
as the network backends for nic devices.

### CODE REVIEW

**Comment by yanglei-rh** [comment]
LGTM and it can test pass on my side.

**Review by yanglei-rh — APPROVED** [review]

File: virttest/qemu_devices/qdevices.py
Comment (PaulYuuu) [inline virttest/qemu_devices/qdevices.py:2445]: I would recommend adding `--foreground` by default, do not let `QDaemonDev` manage a background process, I think it's outside the scope of its design. And with this, `process.pid_exists(pid)` may not a must, can just use the original one.

But let's listen to other peoples' voices.

File: virttest/qemu_devices/qdevices.py
Comment (PaulYuuu) [inline virttest/qemu_devices/qdevices.py:2545]: Can this be covered in `elif isinstance(value, str) and value.isdigit():`?

**Comment by yanglei-rh** [comment]
Hello @luckyh This patch can test pass from my side, and looks like there are also no other issues, can it be merge?

**Comment by luckyh** [comment]
@yanglei-rh according to the PR review workflow of VT we just need another approval for this one before getting ready for merge, so let's wait for @PaulYuuu 's feedback, thanks!

File: virttest/qemu_devices/qdevices.py
Comment (PaulYuuu) [reply virttest/qemu_devices/qdevices.py:2445]: Put https://github.com/avocado-framework/avocado-vt/pull/2929 as a reference. In qdevices.py, we already use aexpect.run_bg, any process will run in backaground, not need to background again.

@yanglei-rh, feel free leave your comment.

Comment (PaulYuuu) [reply virttest/qemu_devices/qdevices.py:2445]: And a unit test like this:
```python
>>> passt1 = client.run_bg("passt")
>>> passt2 = client.run_bg("passt -f")
>>> passt1.is_alive()
False
>>> passt2.is_alive()
True
```

Comment (yanglei-rh) [reply virttest/qemu_devices/qdevices.py:2445]: Hi @PaulYuuu Sorry for the late reply, I agree with your point.

File: virttest/qemu_devices/qdevices.py
Comment (luckyh) [reply virttest/qemu_devices/qdevices.py]: Good catch! have updated, thanks!

File: virttest/qemu_devices/qdevices.py
Comment (yanglei-rh) [reply virttest/qemu_devices/qdevices.py:2443]: Hi @PaulYuuu Thanks for pointing it out
Hi @luckyh Maybe we can directly to use `aexpect.run_bg` instead of call `QDaemonDev`?

Comment (luckyh) [reply virttest/qemu_devices/qdevices.py:2443]: As daemonized processes are always configured with extra settings [1], I'd suggest keeping the daemon mode in favor of problem detection, in case the user made SELinux enforced.

What do you think @PaulYuuu ?

[1] https://passt.top/passt/tree/util.c#n449

**Review by PaulYuuu — APPROVED** [review]

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Pull request merged**
